src/borg/archiver.py

```diff
@@ -1517,6 +1518,41 @@ def do_with_lock(self, args, repository):
             # see issue #1867.
             repository.commit()
 
+    @with_repository(exclusive=True, cache=True, compatibility=(Manifest.Operation.WRITE,))
+    def do_config(self, args, repository, manifest, key, cache):
+        """get, set, and delete values in a repository or cache config file"""
+        try:
+            section, name = args.name.split('.')
+        except ValueError:
+            section = args.cache and "cache" or "repository"
+            name = args.name
```

Comment:
the above could be a argparse validator and it could be more strict, always requiring section.name to avoid any ambiguities and future problems. it would not need a `--cache` option then, because it is cache.*.

hmm, while thinking about it, isn't it rather configtype:sectionname:key?

configtype = repository or cache
sectionname = [section] from ini-like file

Reply:
90% of the stuff in the config file is in the main section (it would be rare that someone would want to edit e.g. `integrity`), it simplifies it to have the default.

Reply:
there's also the issue that section and configtype are not kept separately.

Reply:
The "default" section (the one with the most important keys in it) is`cache` for the cache config and `repository` for the repo config.
Most commands with a similar purpose I know of (`git config`, `npm config`, etc.) use the same syntax as here, with a --flag for the non-default location.

Reply:
ok, keep it as is.